exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillMergeProjectRule.java

```diff
@@ -48,6 +50,12 @@ public boolean matches(RelOptRuleCall call) {
     Project topProject = call.rel(0);
     Project bottomProject = call.rel(1);
 
+    // Make sure both projects be LogicalProject.
+    if (topProject.getTraitSet().getTrait(ConventionTraitDef.INSTANCE) != Convention.NONE ||
```

Comment:
At first glance, it is non-intuitive that a Drill specific MergeProjectRule returns False if a DRILL_LOGICAL convention is encountered.  However, currently, this rule only checks for the complex expression (e.g convert_to_json), so your change is fine in the context of the current rule.  
In future, would we consider allowing merging 2 DrillProjectRels ? (I think yes).